The Big View/Controller/Resource Rewrite

seancribbs opened the pull request:
This huge PR represents several weeks of trying to make the app and its resources more friendly to Ember.  Some things we did:
- The join-relationship between the `Test` and `Scorecard` entities is now enshrined in a `TestInstance` entity/model/resource. The models have been greatly simplified as a result.
- We don't create the matrix entries ahead of time (bye-bye `ScorecardSubcell`), but instead derive them from the contents of the scorecard's test instances. This is cleaner and results in slightly faster rendering of the matrix.
- The caching adapter is disabled for now, but the REST adapter now fetches in batches, reducing pauses and errors caused by URIs that are too long for the browser.
- Ember and Ember-Data have been updated.
- Lots of JS bugs/style issues have been fixed with `jshint`.

Miscellaneous other stuff:
- The font files are downloaded now with the proper media type.

CC: @cmeiklejohn @joedevivo @coderoshi

cmeiklejohn committed:
Update ember and ember-data.

cmeiklejohn committed:
Use current revision, change transforms syntax.

cmeiklejohn committed:
Define the serializer on the adapter.

cmeiklejohn committed:
Properly map keys.

cmeiklejohn committed:
Fix dat lint.

Commit:
Use mappings to populate some of the associations.

cmeiklejohn committed:
Reference the project correctly.

cmeiklejohn committed:
Load tests by id.

Commit:
Add resources and serializers for tests.

Commit:
Fix dem test_ids

cmeiklejohn committed:
Defer rendering until the grid is ready.

Commit:
WIP: Disentangle model stuff.

Commit:
WIP: Scorecard working with bindings

Commit:
When fetching a test, return false if find raised an exception.

Commit:
WIP adding more functionality

Commit:
Create the TestInstance dummy model/resource so we can cleanup the relationships on the Ember side.

Commit:
Modify models to match what the API is producing.

Commit:
Major restructuring, almost there.

Commit:
Some cleanup, add pretty urls.

Commit:
Update ember-data and ember.

Commit:
Add .woff to the known MIME_TYPES

Commit:
resource_exists should return false, not throw an exception

Commit:
Fix adapter, adding batch fetches.

Commit:
Remove some console.log from ember-data.js.

Commit:
Fix case in URL generation where tagString is undefined/null.

Commit:
Expose the test attributes on the test-instance model instead.

Commit:
Fix TestInstance.tagString.

Commit:
Update to latest ember and add matrix stuff.

It is still horridly slow because we seem to be computing the matrix
too soon.

Commit:
Remove ember-matrix, it's too slow. But doing it manually seems snappy enough.

Commit:
Final cleanup, sgb

Commit:
Fix some jshint stuff, add a task for it.

Commit:
Fix some adapter issues and update dem libs.

* Update Handlebars and Data.
* Move IndexedDB stuff to the side.
* Assign the serializer properly so we can get the right association
  keys.
* Fix deprecation warning in router.

coderoshi committed:
Exit at missing jshint, multiply percent by 100

coderoshi commented on assets/javascripts/app/views.js:
```diff
@@ -1,6 +1,9 @@
 GiddyUp.prettyDate = function(date){
   // Based roughly on distance_of_time_in_words from Ruby on Rails,
   // with ideas from John Resig's "prettyDate" function.
+  if(date === null || date === undefined)
```
shouldn't that be `typeof(date) === 'undefined'`, or is that old school?

seancribbs replied:
Actually `Ember.isNone(date)` would be better.

coderoshi commented:
Awaiting the decrease in fetch batch size, and a reduction in requests, and presuming that the CachingAdapter isn't supposed to be used, +1

Commit:
Switch to rainbows, decrease batch size, fix a few style issues and bugs.

* Puma dislikes being hammered with too many requests and will 500 if
  it reaches a limit. Rainbows with the FiberPool strategy seems to
  work nicely.
* Decrease the batch size for extra safety.
* Use -- instead of -=.
* Sometimes when switching scorecards the dimensionValues wouldn't
  have changed before recomputing the matrix. Making them volatile has
  little cost and keeps them nice and fresh.

Commit:
Add db dump from 2012-12-24 for dev/testing purposes.

seancribbs commented:
@coderoshi Thanks, merging and deploying. YAY

cmeiklejohn commented:
Nice work, @seancribbs!
